ranking.php

```diff
@@ -36,62 +36,58 @@
 require_login();
 $context = context_system::instance();
 require_capability("block/coursefeedback:managefeedbacks", $context);
-$action = optional_param("action", null, PARAM_ALPHA);
 
-if ($action == "download")
-{
-    $feedbackid = required_param("feedback",  PARAM_INT);
+$config = get_config("block_coursefeedback");
+$PAGE->set_url(new moodle_url("/blocks/coursefeedback/ranking.php"));
+$PAGE->set_context($context);
 
-    // Get the questionid for the feedback from the DB-id of the question.
-    $qufbid = null;
-    if ($qudbid = optional_param("question", null, PARAM_INT)) {
-        $qufb = $DB->get_record("block_coursefeedback_questns", array("id" => $qudbid));
-        $qufbid = $qufb->questionid;
-    }
+$form = new f_ranking_form();
+if ($form->is_submitted() && $form->is_validated()) {
+    $data = $form->get_data();
 
-    $lang = current_language();
-    $export = new rankingexport($feedbackid, $qufbid);
-    if (true) //$export->init_format($download))
-    {
-        $filename = "RANKING" //get_string("download_html_filename", "block_coursefeedback")
-            . date("_Y-m-d-H-i")
-            . ".csv";
-        $export->create_file($lang);
-        header("Content-Type: text/csv");
-        header("Content-Disposition: attachment; filename=" . $filename);
-        echo $export->get_content();
+    // Checking validation here again because JS validation (show/hide downloadbuttons might not be working
+    // Standard validation in form f_ranking_form was not working because the warnigs would not dissapear
+    // because through download the page is not new loading
+    if (!isset($data->feedback) || $data->feedback == 0) {
```

Comment:
`empty()` tut genau das (probiers am besten nochmal aus). Siehe auch folgende Zeilen
```suggestion
    if (empty($data->feedback)) {
```